PhysicsTools/PatAlgos/plugins/PATJetUpdater.cc

```diff
@@ -0,0 +1,144 @@
+//
+
+
+#include "PhysicsTools/PatAlgos/plugins/PATJetUpdater.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Common/interface/Association.h"
+#include "DataFormats/Candidate/interface/CandAssociation.h"
+
+#include "DataFormats/PatCandidates/interface/JetCorrFactors.h"
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "FWCore/Utilities/interface/transform.h"
+
+#include <vector>
+#include <memory>
+#include <algorithm>
+
+
+using namespace pat;
+
+
+PATJetUpdater::PATJetUpdater(const edm::ParameterSet& iConfig) :
+  useUserData_(iConfig.exists("userData"))
+{
+  // initialize configurables
+  jetsToken_ = consumes<edm::View<Jet> >(iConfig.getParameter<edm::InputTag>( "jetSource" ));
+  addJetCorrFactors_ = iConfig.getParameter<bool>( "addJetCorrFactors" );
+  jetCorrFactorsTokens_ = edm::vector_transform(iConfig.getParameter<std::vector<edm::InputTag> >( "jetCorrFactorsSource" ), [this](edm::InputTag const & tag){return mayConsume<edm::ValueMap<JetCorrFactors> >(tag);});
```

Comment:
whether or not the ValueMap<JetCorrFactors> will be consumed is known according to the value of addJetCorrFactors_. Correct? if so, change the mayConsume to consume (within an if statement on the value of addJetCorrFactors_)